src/include/storage/buffer_manager/bm_file_handle.h

```diff
@@ -46,13 +46,42 @@ class PageState {
     std::atomic<uint64_t> evictionTimestamp = 0;
 };
 
+// This class is used to keep the WAL page idxes of a page group in the original file handle.
+// For each original page in the page group, it has a corresponding WAL page idx. The WAL page idx
+// is initialized as INVALID_PAGE_IDX to indicate that the page does not have any updates in WAL
+// file. To synchronize accesses to the WAL version page idxes, we use a lock for each WAL page idx.
+class WALPageIdxGroup {
+public:
+    WALPageIdxGroup();
+
+    inline void acquireWALPageIdxLock(common::page_idx_t originalPageIdxInGroup) {
```

Comment:
Might be good to clarify what originalPageIdxInGroup means with an example. As I understand it is the offset of the page in the pageIdxGroup, right?

Reply:
Right. Added a comment.